packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_CoarseOperator_def.hpp

```diff
@@ -664,6 +734,158 @@ namespace FROSch {
                 }
 #endif
             }
+        } else if(!DistributionList_->get("Type","linear").compare("ZoltanDual")){
+          //ZoltanDual provides a factorization of the coarse problem with Zoltan2 inlcuding the
+          //build of a Repeated map suited for the next level
+          //GatheringSteps to communicate Matrix
+          int gatheringSteps = DistributionList_->get("GatheringSteps",1);
+          GatheringMaps_.resize(gatheringSteps);
+          CoarseSolveExporters_.resize(gatheringSteps);
+
+          double gatheringFactor = pow(double(this->MpiComm_->getSize())/double(NumProcsCoarseSolve_),1.0/double(gatheringSteps));
+          LO numProcsGatheringStep = this->MpiComm_->getSize();
+          GO numGlobalIndices = CoarseMap_->getMaxAllGlobalIndex();
+          GO numMyRows;
+          numMyRows = 0;
+
+          if (this->MpiComm_->getRank()%(this->MpiComm_->getSize()/NumProcsCoarseSolve_) == 0 && this->MpiComm_->getRank()/(this->MpiComm_->getSize()/NumProcsCoarseSolve_) < NumProcsCoarseSolve_) {
+            if (this->MpiComm_->getRank()==0) {
+                numMyRows = numGlobalIndices - (numGlobalIndices/NumProcsCoarseSolve_)*(NumProcsCoarseSolve_-1);
+            } else {
+                numMyRows = numGlobalIndices/NumProcsCoarseSolve_;
+            }
+          }
+
+          XMapPtr tmpCoarseMap = Xpetra::MapFactory<LO,GO,NO>::Build(CoarseMap_->lib(),-1,numMyRows,0,this->MpiComm_);
+          if (tmpCoarseMap->getNodeNumElements()>0) {
+              OnCoarseSolveComm_=true;
+          }
+          CoarseSolveComm_ = this->MpiComm_->split(!OnCoarseSolveComm_,this->MpiComm_->getRank());
+
+          //Gathering Steps for RepeatedMap#################################################
+          //-> Have to test that
```

Comment:
tested?

Reply:
I now tested the gathering steps and had to make some changes. Now it works fine.